### PR TITLE
Missile Function Names and Enums Cleanup

### PR DESCRIPTION
**Pull request by kphoenix137**
Also cleans up some of spelldat.cpp for consistency with misdat.cpp

### CODE REVIEW

File: Source/itemdat.cpp [hunk 1]
```diff
@@ -141,29 +141,29 @@ ItemData AllItemsList[] = {
...
-/*               */ {  IDROP_REGULAR, ICLASS_MISC,   ILOC_UNEQUIPABLE, ICURS_SCROLL_OF,                   ItemType::Misc,        UITYPE_NONE,        N_("Scroll of Chain Lightning"),   nullptr,          10,           0,       0,       0,      0,      0,       0,      35,       0, ISPL_NONE,         IMISC_SCROLLT,   SPL_CHAIN,       true,       750 },
+/*               */ {  IDROP_REGULAR, ICLASS_MISC,   ILOC_UNEQUIPABLE, ICURS_SCROLL_OF,                   ItemType::Misc,        UITYPE_NONE,        N_("Scroll of Stone Curse"),       nullptr,           6,           0,       0,       0,      0,      0,       0,      33,       0, ISPL_NONE,         IMISC_SCROLLT,   SPL_STONECURSE,  true,       800 },
+/*               */ {  IDROP_REGULAR, ICLASS_MISC,   ILOC_UNEQUIPABLE, ICURS_SCROLL_OF,                   ItemType::Misc,        UITYPE_NONE,        N_("Scroll of Chain Lightning"),   nullptr,          10,           0,       0,       0,      0,      0,       0,      35,       0, ISPL_NONE,         IMISC_SCROLLT,   SPL_CHAINLIGHTNING,true,       750 },
```
Comment (qndel) [inline Source/itemdat.cpp:159]: broken alignment

Comment (kphoenix137) [reply Source/itemdat.cpp:159]: Will fix tomorrow :)

File: Source/missiles.h [hunk 2]
```diff
@@ -215,22 +215,22 @@ void AddStealPotions(Missile &missile, const AddMissileParameter &parameter);
 void AddManaTrap(Missile &missile, const AddMissileParameter &parameter);
 void AddSpecArrow(Missile &missile, const AddMissileParameter &parameter);
 void AddWarp(Missile &missile, const AddMissileParameter &parameter);
-void AddLightningWall(Missile &missile, const AddMissileParameter &parameter);
+void AddLightningwall(Missile &missile, const AddMissileParameter &parameter);
```
Comment (qndel) [inline Source/missiles.h:218]: why

File: Source/missiles.h [hunk 3]
```diff
@@ -215,22 +215,22 @@ void AddStealPotions(Missile &missile, const AddMissileParameter &parameter);
...
 void AddRuneExplosion(Missile &missile, const AddMissileParameter &parameter);
-void AddFireNova(Missile &missile, const AddMissileParameter &parameter);
-void AddLightningArrow(Missile &missile, const AddMissileParameter &parameter);
+void AddImmolation(Missile &missile, const AddMissileParameter &parameter);
+void AddBowLightning(Missile &missile, const AddMissileParameter &parameter);
 void AddMana(Missile &missile, const AddMissileParameter &parameter);
 void AddMagi(Missile &missile, const AddMissileParameter &parameter);
-void AddRing(Missile &missile, const AddMissileParameter &parameter);
+void AddRingoffire(Missile &missile, const AddMissileParameter &parameter);
```
Comment (qndel) [inline Source/missiles.h:224]: no

File: Source/missiles.h (hunk 2 above)
Comment (kphoenix137) [reply Source/missiles.h:218]: Followed the format that was more consistently used where only the first letter used in the spell name was capitalized regardless of how many words are in the spell name. I can capitalize first letter of all words if it looks better

File: Source/missiles.h (hunk 3 above)
Comment (kphoenix137) [reply Source/missiles.h:224]: RingOfFire? RingFire? FireRing?

File: Source/player.cpp [hunk 4]
```diff
@@ -1259,12 +1259,12 @@ bool DoRangeAttack(int pnum)
 		}
 
 		int dmg = 4;
-		missile_id mistype = MIS_ARROW;
+		missile_id mistype = MIS_ACID;
```
Comment (qndel) [inline Source/player.cpp:1262]: what happened here

Comment (kphoenix137) [reply Source/player.cpp:1262]: mistake

**Comment by AJenbo** [comment]
Looks like this all got replaced by https://github.com/diasurgical/devilutionX/pull/5733 and SpellID

**Pull request closed without merging**